yarn-project/simulator/src/private/simulator.ts

```diff
@@ -42,15 +41,15 @@ export class AcirSimulator {
     msgSender = AztecAddress.fromField(Fr.MAX_FIELD_VALUE),
     scopes?: AztecAddress[],
   ): Promise<PrivateExecutionResult> {
-    const header = await this.db.getBlockHeader();
+    const header = await this.executionDataProvider.getBlockHeader();
 
     await verifyCurrentClassId(
       contractAddress,
-      await this.db.getContractInstance(contractAddress),
-      this.node,
+      await this.executionDataProvider.getContractInstance(contractAddress),
+      this.executionDataProvider,
```

Comment:
This function is weird because it's vomiting complexity to its args - the instance arg is completely unnecessary. Would change it such that it doesn't take the instance and instead it obtains it in the `verifyCurrentClassId` function body.

I see that this got uncovered with your changes as before the node and db were separate things. This seems like a good sign.

Reply:
You're right, updated. I also tried to fix `readCurrentClassId` but unfortunately it's hard to make pretty without retrieving the contract instance again, which is repeating work and can lead to nasty bugs.